sphinx/themes/basic/static/searchtools.js

```diff
@@ -328,7 +329,7 @@ const Search = {
     for (const [title, foundTitles] of Object.entries(allTitles)) {
       if (title.toLowerCase().trim().includes(queryLower) && (queryLower.length >= title.length/2)) {
         for (const [file, id] of foundTitles) {
-          let score = Math.round(100 * queryLower.length / title.length)
+          let score = Math.round((titles[file] !== title ? Scorer.title : Scorer.mainTitle) * queryLower.length / title.length)
```

Comment:
I think I preferred my approach of creating a temporary called `isMainTitle` and then reusing it both here and below, but it's not that big a deal.

Reply:
Let me see if I can find a middle-ground (that might also resolve the overloads-backwards-compat concern).

Reply:
I'll push one more variation, to see what you think.  I want to make it as clear as possible that there are essentially two changes we're making here: (1) changing the base score from `100` to `Scorer.title`, and (2) adding a boost for main document titles.